soc/realtek/ec/rts5912/reg/reg_gpio.h

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2023 Realtek, SIBG-SD7
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * @brief GPIO Controller (GPIO)
+ */
+
+typedef struct {
+
+	union {
+		volatile uint32_t GCR[132];
+
+		struct {
+			volatile uint32_t DIR			: 1;
+			volatile uint32_t INDETEN		: 1;
+			volatile uint32_t INVOLMD		: 1;
+			volatile uint32_t PINSTS		: 1;
+			uint32_t reserved1				: 4;
+			volatile uint32_t MFCTRL		: 3;
+			volatile uint32_t OUTDRV		: 1;
+			volatile uint32_t SLEWRATE		: 1;
+			volatile uint32_t PULLDWEN		: 1;
+			volatile uint32_t PULLUPEN		: 1;
+			volatile uint32_t SCHEN			: 1;
+			volatile uint32_t OUTMD			: 1;
+			volatile uint32_t OUTCTRL		: 1;
+			uint32_t reserved2				: 6;
+			volatile uint32_t INTCTRL		: 3;
+			uint32_t reserved3				: 1;
+			volatile uint32_t INTEN			: 1;
+			uint32_t reserved4				: 2;
+			volatile uint32_t INTSTS		: 1;
+		} GCR_b[132];
+	};
+} GPIO_Type;
+
+/* GCR */
+#define GPIO_GCR_DIR_Pos					(0UL)
+#define GPIO_GCR_DIR_Msk					(0x1UL)
```

Comment:
You can use `GENMASK(h,l)` for all these mask fields.  This makes it easier to match the macros to the datasheet.

Reply:
We use GENMASK(h,l) and BIT() to replace the mask fields of the files which is in the soc\realtek\ec\rts5912\reg .